src/QsCompiler/QirGeneration/RewriteSteps.cs

```diff
@@ -74,8 +74,17 @@ public bool PreconditionVerification(QsCompilation compilation)
         /// <inheritdoc/>
         public bool Transformation(QsCompilation compilation, out QsCompilation transformed)
         {
+            bool debugSymbolsEnabled = false;
+            if (this.AssemblyConstants.TryGetValue(ReservedKeywords.AssemblyConstants.DebugSymbolsEnabled, out string? debugEnabledString))
```

Comment:
Is this a workaround to pass whether debug symbols are enabled to the QirGeneration rewrite step?

Reply:
That's what it's for, but I don't think it's a workaround. I think this is the way the assembly constants are meant to be used. Shortly after this the Qir Output Path is received using the same method. This just needs to occur earlier because it is relevant whether debug symbols are enabled before the generator is applied whereas the output path is only relevant after the generator is applied and we have the content of the QIR.

Reply:
Sounds good, thanks for the clarification Ryan!